### PR TITLE
Generic book description

### PR DESCRIPTION
**Pull request by BrynL-F**
Needs a lil' refactoring. Also image carousel/image not done for book.

### CODE REVIEW

File: app/src/main/java/com/stl/skipthelibrary/ViewBookActivity.java [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        rating_element.setMax(book.getRating().getMaxRating());
+        rating_element.setNumStars((int) Math.round(book.getRating().getAverageRating()));
+        synopsis_element.setText(book.getDescription().getSynopsis());
+    }
+
+    private void setBookDescriptionFieldsEditable(Boolean isEditable) {
+        if (isEditable) {
+            title_element.setEnabled(true);
+            author_element.setEnabled(true);
+            rating_element.setEnabled(true);
```
Comment (gillianpierce) [inline app/src/main/java/com/stl/skipthelibrary/ViewBookActivity.java]: Rating will be pulled from an api so it should never be editable by the user 
```suggestion
```

**Comment by tymoorej** [comment]
Branch may need to be nuked. Probably best to create a new branch off of master and copy paste all relevant changes.

**Comment by tymoorej** [comment]
> Branch may need to be nuked. Probably best to create a new branch off of master and copy paste all relevant changes.

@LukeSlev any objections?

File: app/src/main/res/layout/book_description.xml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        <EditText
+            android:id="@+id/title_element"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/title"
+            android:textSize="24dp"
+            android:textColor="@color/white"
+            android:fontFamily="@font/poppins"
+            app:layout_constraintTop_toTopOf="@id/corners"
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]: This view should probably be constraint at start and end as well 
```suggestion
            app:layout_constraintTop_toTopOf="@id/corners"
            app:layout_constraintStart_toStartOf="parent"
            app:layout_constraintEnd_toEndOf="parent"
```

File: app/src/main/res/layout/book_description.xml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+
+        <EditText
+            android:id="@+id/title_element"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/title"
+            android:textSize="24dp"
+            android:textColor="@color/white"
+            android:fontFamily="@font/poppins"
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]: I think poppins_semibold looks better for headings
```suggestion
            android:fontFamily="@font/poppins_semibold"
```

File: app/src/main/res/layout/book_description.xml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginVertical="8dp"
+            android:text="@string/rating"
+            app:layout_constraintTop_toBottomOf="@id/book_image"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            android:textSize="18sp"
+            android:textColor="@color/white"
+            android:fontFamily="@font/poppins"
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]:
```suggestion
            android:fontFamily="@font/poppins_semibold"
```

File: app/src/main/res/layout/book_description.xml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            android:background="@color/white"
+            android:layout_marginTop="16dp"
+            app:layout_constraintTop_toBottomOf="@+id/rating_bar_element" />
+
+
+        <TextView
+            android:id="@+id/book_details_heading"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toBottomOf="@id/rating_bar_element"
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]: This view should also have start and end constraints
```suggestion
            app:layout_constraintTop_toBottomOf="@id/rating_bar_element"
            app:layout_constraintEnd_toEndOf="parent"
            app:layout_constraintStart_toStartOf="parent"
```

File: app/src/main/res/layout/book_description.xml [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+            android:fontFamily="@font/poppins"
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]:
```suggestion
            android:fontFamily="@font/poppins_semibold"
```

File: app/src/main/res/layout/book_description.xml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            android:text="@string/details"
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]: I think description or synopsis would be a better heading here
```suggestion
            android:text="@string/description"
```

File: app/src/main/res/layout/book_description.xml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+            android:textColor="@color/white" />
```
Comment (gillianpierce) [inline app/src/main/res/layout/book_description.xml]: I noticed when running this branch that the bottom dividing line was intersecting this text element, A top and right margin should make it look better! Also text size 18sp differentiates it from the body text :)
```suggestion
            android:textColor="@color/white" 
            android:layout_marginStart="8dp"
            android:layout_marginTop="16dp"
            android:textSize="18sp" />
```

**Pull request closed without merging**